Sources/NiceComponents/Button/NiceButton.swift

```diff
@@ -77,13 +90,13 @@ public struct NiceButton: View {
                        weight: style.textStyle.weight,
                        maxSize: style.textStyle.dynamicTypeMaxSize
                    )
-                   .padding(.leading, leftImage?.offset ?? 0)
-                   .padding(.trailing, rightImage?.offset ?? 0)
+                   .padding(.leading, leftImage?.offset ?? textPadding ?? 0)
+                   .padding(.trailing, rightImage?.offset ?? textPadding ?? 0)
                if let rightImage = rightImage {
                    rightImage.image
                }
            }
-           .frame(maxWidth: .infinity)
+           .frame(maxWidth: maxWidth)
```

Comment:
I think we can get rid of the `maxWidth` parameter with the same result. We just have to add `nil` as the default value of `textPadding`.
![image](https://github.com/user-attachments/assets/b44f5040-a39b-43fc-ac8f-b7fe44889cc7)
```suggestion
           .frame(maxWidth: horizontalPadding == nil ? .infinity : nil)
```